spec.bs

```diff
@@ -233,6 +238,52 @@ The <dfn for="interest group">estimated size</dfn> of an [=interest group=] |ig|
     [=interest group ad/render url=].
   1. The [=string/length=] of |ad|'s [=interest group ad/metadata=] if the field is not null.
 
+<div algorithm>
+
+To <dfn>check interest group permissions</dfn> given an [=origin=]
+|ownerOrigin|, an [=origin=] |frameOrigin|, and a [=boolean=] |isJoin|:
+1. If |ownerOrigin| is [=same origin=] to |frameOrigin|, then return true.
+1. Let |permissionsUrl| be the result of [=building an interest group permissions url=]
+  with |ownerOrigin| and |frameOrigin|.
+1. Let |request| be the result of [=creating a request=] with |permissionsUrl|, [=create a request/accept=]
+  set to "`application/json`"|. TODO: [maybe](https://source.chromium.org/chromium/chromium/src/+/main:content/browser/interest_group/interest_group_permissions_checker.cc;drc=adac219925ef5d9c9a954d189c2e4b8852a4bbed;l=123) require cors mode here?
+1. Let |resource| be null.
+1. [=Fetch=] |request| with [=fetch/processResponseConsumeBody=] set to the following steps given a
+  [=response=] |response| and |responseBody|:
+  1. If |responseBody| is null or failure, set |resource| to failure and return.
+  1. Let |headers| be |response|'s [=response/header list=].
+  1. If [=header list/getting a structured field value=] "X-Allow-FLEDGE" from |headers| does not
+    return true, set |resource| to failure and return.
+  1. Let |mimeType| be the result of [=header list/extracting a MIME type=] from |headers|.
+  1. If |mimeType| is failure or is not a [=JSON MIME Type=], throw, set |resource| to failure and return.
+  1. Set |resource| to |responseBody|.
+1. Wait for |resource| to be set.
+1. If |resource| is failure, then return false.
+1. Let |permissions| be the result of [=parsing a JSON string to an Infra value=], returning false on failure.
+1. If |permissions| is not an [=ordered map=], then return false.
+1. If |isJoin| is true and |permissions|["`joinAdInterestGroup`"] [=map/exists=], then return |permissions|["`joinAdInterestGroup`"].
+1. If |isJoin| is false and |permissions|["`leaveAdInterestGroup`"] [=map/exists=], then return |permissions|["`leaveAdInterestGroup`"].
+1. Return false.
+
+The browser may cache requests for |permissionsUrl| within a network partition.
+
+In order to prevent leaking data, the browser must request |permissionsUrl|
+regardless of whether the user is a member of the ad interest group. This
+prevents a leak of the user's ad interest group membership to the server.
+
+</div>
+
+<div algorithm>
+
+To <dfn>build an interest group permissions url</dfn> given a [=origin=] |ownerOrigin| and a [=origin=] |frameOrigin|:

```

Comment:
(not for right now, but we could probably refactor some of the query params stuff using https://url.spec.whatwg.org/#interface-urlsearchparams)